### ISSUE TITLE
FluxStore version 1.4.1: Missing Reviews

### ISSUE THREAD

**Issue by kalakgl** [inspireui/support#3474]
This latest version don't show the reviews in the app. This was working on previous version.

**Comment by philipvu30** [comment]
Hi,

Currently, the app is compatible with these versions.

ListingPro

- Theme version 2.5.1
- ListingPro Plugin version 2.5.1
- ListingPro Review 1.2

My Listing
- Theme version 2.2.2

Listeo
- Theme version 1.3.1

And the latest Listapp Mobile Manager plugin.

**Comment by kalakgl** [comment]
Hi
I am running Listeo version 1.3.3

**Comment by kalakgl** [comment]
![review_1](https://user-images.githubusercontent.com/12363496/83428073-87067200-a410-11ea-8182-1db55c4715fc.jpg)
![review_2](https://user-images.githubusercontent.com/12363496/83428074-879f0880-a410-11ea-8da4-67ec4ec5106b.jpg)

**Comment by kalakgl** [comment]
You closed my "log in" ticket. Now what, it still don't work :-)

**Comment by philipvu30** [comment]
Hi,

Users are recommended to keep all the issues in one ticket when submitting verification form at the very first line. So for further questions, please post it here.

Back to your issue, as I stated before, your app now is compatible with Listeo 1.3.1 theme.
We haven't followed up to the Listeo version 1.3.3 yet.

**Comment by kalakgl** [comment]
Well I updated from 1.2.9 to 1.3.3 just to be sure. So it didn't Work in 1.2.9 on this new version.

**Comment by philipvu30** [comment]
Hi,

Regarding the review issue, in your config.dart, please change this variable from lp-reviews to getReviews.

const kListingReviewMapping = {
  "review": "lp-reviews",
  "item": "lp_listingpro_options.listing_id"
};

**Comment by kalakgl** [comment]
I changed it to this:
const kListingReviewMapping = {
  "review": "getReviews",
  "item": "lp_listingpro_options.listing_id"
};
Now I get this error:
<img width="328" alt="Skærmbillede 2020-06-02 kl  08 53 32" src="https://user-images.githubusercontent.com/12363496/83512188-9fc56500-a4ae-11ea-9f99-487b559f5ba2.png">

**Comment by philipvu30** [comment]
Hi,

In order for the reviews to display, the comment must be approved. If you don't want this kind of behavior, please remove the if (review.status == 'approved') in lib/services/listing.dart.

Thank you.

![image](https://user-images.githubusercontent.com/54442328/83598854-0329c280-a595-11ea-9ac8-e86585b7cb99.png)

**Comment by kalakgl** [comment]
The review is approved Nd visible online

Ulrik Motzfeldt-Skovgaard
Sendt fra min iPhone

Den 3. jun. 2020 kl. 03.23 skrev philipvu30 <notifications@github.com>:

﻿

Hi,

In order for the reviews to display, the comment must be approved. If you don't want this kind of behavior, please remove the if (review.status == 'approved') in lib/services/listing.dart.

Thank you.

[image]<https://user-images.githubusercontent.com/54442328/83598854-0329c280-a595-11ea-9ac8-e86585b7cb99.png>

**Comment by philipvu30** [comment]
Hi,

I would like to give you my apologies. It was a mistake that we uploaded the wrong version of ListApp Mobile Manager plugin.
Please manually delete the plugin on your website and upload this [plugin](https://github.com/inspireui/support/files/4722691/listapp-mobile-manager.zip) instead. 

Thank you for your patience.

**Comment by kalakgl** [comment]
Hi

☺

I was already using 1.6.3, but I deleted manually and uploaded this newly downloaded file.
But sorry still the same problem with “no reviews” on the new app.


Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard

Fra: philipvu30 <notifications@github.com>
Sendt: onsdag den 3. juni 2020 08.38
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>; Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

I would like to give you my apologies. It was a mistake that we uploaded the wrong version of ListApp Mobile Manager plugin.
Please manually delete the plugin on your website and upload this plugin<https://github.com/inspireui/support/files/4722691/listapp-mobile-manager.zip> instead.

Thank you for your patience.

**Comment by philipvu30** [comment]
Hi,

If it still not solves your issue, please proceed to remove this line along with the bracket under it.

Thanks.

> Hi,
> 
> In order for the reviews to display, the comment must be approved. If you don't want this kind of behavior, please remove the if (review.status == 'approved') in lib/services/listing.dart.
> 
> Thank you.
> 
> ![image](https://user-images.githubusercontent.com/54442328/83598854-0329c280-a595-11ea-9ac8-e86585b7cb99.png)

**Comment by kalakgl** [comment]
Hi

Sorry but my files seems to be different than yours.

My line 266:

[cid:image001.png@01D639AB.F71F41E0]

Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard



Fra: philipvu30 <notifications@github.com>
Svar til: inspireui/support <reply@reply.github.com>
Dato: onsdag den 3. juni 2020 kl. 13.19
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>, Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

If it still not solves your issue, please proceed to remove this line along with the bracket under it.

Thanks.

Hi,

In order for the reviews to display, the comment must be approved. If you don't want this kind of behavior, please remove the if (review.status == 'approved') in lib/services/listing.dart.

Thank you.

[Billede fjernet af afsender. image]<https://user-images.githubusercontent.com/54442328/83598854-0329c280-a595-11ea-9ac8-e86585b7cb99.png>

**Comment by kalakgl** [comment]
Maybe these lines:

[cid:image001.png@01D639AC.4A953960]

Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard



Fra: philipvu30 <notifications@github.com>
Svar til: inspireui/support <reply@reply.github.com>
Dato: onsdag den 3. juni 2020 kl. 13.19
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>, Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

If it still not solves your issue, please proceed to remove this line along with the bracket under it.

Thanks.

Hi,

In order for the reviews to display, the comment must be approved. If you don't want this kind of behavior, please remove the if (review.status == 'approved') in lib/services/listing.dart.

Thank you.

[image]<https://user-images.githubusercontent.com/54442328/83598854-0329c280-a595-11ea-9ac8-e86585b7cb99.png>

**Comment by philipvu30** [comment]
Hi,

Please replace these files in lib/services. Thank you
[SERVICES.zip](https://github.com/inspireui/support/files/4727259/SERVICES.zip)

**Comment by kalakgl** [comment]
Hi
Thanks,  just replaced with your zip file.
Now I get this:

 [cid:image001.png@01D63A56.6CEBA7B0]

Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard



Fra: philipvu30 <notifications@github.com>
Svar til: inspireui/support <reply@reply.github.com>
Dato: torsdag den 4. juni 2020 kl. 01.06
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>, Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

Please replace these files in lib/services. Thank you
SERVICES.zip<https://github.com/inspireui/support/files/4727259/SERVICES.zip>

**Comment by philipvu30** [comment]
Hi,

I've sent you the latest project via your email. Please check.

**Comment by kalakgl** [comment]
Hi

Thank you, I got the files.

I am using the new files. And I can see the reviews now, but no images, descriptions etc. for an item.


Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard



Fra: philipvu30 <notifications@github.com>
Svar til: inspireui/support <reply@reply.github.com>
Dato: fredag den 5. juni 2020 kl. 00.24
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>, Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

I've sent you the latest project via your email. Please check.

**Comment by michaelbrinkworth** [comment]
join :) https://www.facebook.com/groups/289288585778355/

**Comment by philipvu30** [comment]
Hi,

Sorry for the delayed response as we rarely work on the weekends. Please check that if you have correctly configured the config.dart?
Or just copy the text in lib/example/listeo/config.dart.txt and paste it in your lib/common/config.dart. And replace only your website information in serverConfig.

Thank you.

**Comment by kalakgl** [comment]
Hi

Before you sent me thes elatest files, then I could see all the content, except for the reviews.


Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard

Fra: philipvu30 <notifications@github.com>
Sendt: mandag den 8. juni 2020 01.03
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>; Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

Sorry for the delayed response as we rarely work on the weekends. Please check that if you have correctly configured the config.dart?
Or just copy the text in lib/example/listeo/config.dart.txt and paste it in your lib/common/config.dart. And replace only your website information in serverConfig.

Thank you.

**Comment by philipvu30** [comment]
Hi,

Please replace this file [config.dart.zip](https://github.com/inspireui/support/files/4745413/config.dart.zip)

**Comment by kalakgl** [comment]
Thank you very much !


Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard

Fra: philipvu30 <notifications@github.com>
Sendt: mandag den 8. juni 2020 08.40
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>; Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

Please replace this file config.dart.zip<https://github.com/inspireui/support/files/4745413/config.dart.zip>

**Comment by kalakgl** [comment]
Hi

I am trying to deactivate language selection under “settings” page.
In previous version I just mark the language part, and then it is not shown anymore.
But in this new updated I get this new filed “Empty….”.
The language selectiong is gone, but replaced with this #”Empty”

How can I deactivate language selection completely ?

 [cid:image001.png@01D63D9C.C0E2A640]
Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard



Fra: philipvu30 <notifications@github.com>
Svar til: inspireui/support <reply@reply.github.com>
Dato: mandag den 8. juni 2020 kl. 08.40
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>, Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

Please replace this file config.dart.zip<https://github.com/inspireui/support/files/4745413/config.dart.zip>

**Comment by philipvu30** [comment]
Hi,

To completely remove it from the app, please comment/delete these lines in lib/screens/settings.dart

![image](https://user-images.githubusercontent.com/54442328/84100753-b92e5a00-aa36-11ea-9913-0bdfef6349be.png)

**Comment by kalakgl** [comment]
Sorry but your mail don’t have image ☺


Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard

Fra: philipvu30 <notifications@github.com>
Sendt: tirsdag den 9. juni 2020 00.52
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>; Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

To completely remove it from the app, please comment/delete these lines in lib/screens/settings.dart

[image]<https://user-images.githubusercontent.com/54442328/84100753-b92e5a00-aa36-11ea-9913-0bdfef6349be.png>

**Comment by kalakgl** [comment]
Hi

Yes I have commented these lines out. But then the app shows this “Empty” as I send you.


Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard



Fra: philipvu30 <notifications@github.com>
Svar til: inspireui/support <reply@reply.github.com>
Dato: tirsdag den 9. juni 2020 kl. 00.51
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>, Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Hi,

To completely remove it from the app, please comment/delete these lines in lib/screens/settings.dart

[image]<https://user-images.githubusercontent.com/54442328/84100753-b92e5a00-aa36-11ea-9913-0bdfef6349be.png>

**Comment by philipvu30** [comment]
Hi,

Sorry, I forgot to say this. But all the images you sent via email, I can't see them. 
Please post them via GitHub website. 

Thank you.

**Comment by kalakgl** [comment]
<img width="1167" alt="Skærmbillede 2020-06-08 kl  13 46 49" src="https://user-images.githubusercontent.com/12363496/84139682-11f9f480-aa2f-11ea-9013-b10786895b6c.png">

**Comment by kalakgl** [comment]
As you can see, the lines are commented out, and then the app shows this "Empty"...

**Comment by philipvu30** [comment]
In lib/common/constants.dart, please remove the line language also.

![image](https://user-images.githubusercontent.com/54442328/84140032-27950b80-aa7b-11ea-9f0d-e7caf38a704c.png)

**Comment by kalakgl** [comment]
Thank you ! Now it is gone.

**Comment by kalakgl** [comment]
I have (iOS) Google Log In, Facebook Login and normal registration login working.
But the Apple Login won't work. I have followed your guides for all.
You can see the database error, telling me I am missing some permissions. But I can't tell from your guide, what I am missing.
Can you try to tell me that?

![IMG_1708](https://user-images.githubusercontent.com/12363496/84166927-5055da80-aa54-11ea-9fcb-18e80c0da40f.PNG)

**Comment by philipvu30** [comment]
Hi,

Please check if you have enabled the option "Apple Sign In" in your Runner and Firebase project yet. You can find them here.

![image](https://user-images.githubusercontent.com/54442328/84221453-7a180b80-aaff-11ea-8ce5-af9389a5c927.png)

![image](https://user-images.githubusercontent.com/54442328/84221549-b2b7e500-aaff-11ea-9aa6-06ec2d6ae66a.png)

Thank you.

**Comment by kalakgl** [comment]
Yes these are OK.

**Comment by philipvu30** [comment]
Hi,

Please check the information in this file ios/GoogleService-Info.plist if it matches with the information on your FireBase project. Because I can use the Apple Sign-in with your config with our FireBase information.

Thank you.

**Comment by kalakgl** [comment]
Yes, I can see you have created an account.
Can I mail you screen shots directly to you, so I don't share it through here.
Have downloaded the latest ios/GoogleService-Info.plist, but still have the same error.

**Comment by philipvu30** [comment]
Sure. Please send it to toan@inspireui.com

**Comment by philipvu30** [comment]
Can I see the rules in cloud firestore?
![image](https://user-images.githubusercontent.com/54442328/84376112-c439e380-ac0a-11ea-9eea-d0c846608543.png)

**Comment by kalakgl** [comment]
Which page is it ?


Inuss. inuull. / Venlig Hilsen / Best Regards,
Ulrik Motzfeldt-Skovgaard

Fra: philipvu30 <notifications@github.com>
Sendt: torsdag den 11. juni 2020 08.42
Til: inspireui/support <support@noreply.github.com>
Cc: Ulrik Motzfeldt-Skovgaard <ulmo@qeqqata.gl>; Author <author@noreply.github.com>
Emne: Re: [inspireui/support] FluxStore version 1.4.1: Missing Reviews (#3474)


Can I see the rules in cloud firestore?
[image]<https://user-images.githubusercontent.com/54442328/84376112-c439e380-ac0a-11ea-9eea-d0c846608543.png>

**Comment by kalakgl** [comment]
<img width="805" alt="Skærmbillede 2020-06-11 kl  08 48 25" src="https://user-images.githubusercontent.com/12363496/84376797-65e81300-abc0-11ea-86b4-0211335355cf.png">

**Comment by kalakgl** [comment]
I just try to delete : if false; and replaced it with ;
So it looks like yours. Didn't helped.

**Comment by philipvu30** [comment]
Would you mind running these commands inside the ios folder?
pod install
pod update

I doubt it is the cause but maybe we'll get lucky.

**Comment by kalakgl** [comment]
Sorry - it didn't helped.

**Comment by philipvu30** [comment]
Are you using Apple Developer 99$ account?

**Comment by kalakgl** [comment]
Yes

**Comment by philipvu30** [comment]
Well.

I don't know what is the problem here. Send me your project after running flutter clean. I'll help you check it tmr.

**Comment by kalakgl** [comment]
Just send the cleaned package through WeTransfer.
Thank you very much !

**Comment by thawhakyi** [comment]
hi @philipvu30 
I am also facing review problem. I used Mylisting Theme. how can I send my purchase code to u?

thank you!

**Comment by philipvu30** [comment]
Hi @kalakgl 

I think I know what is your problem is. Please head to FireBase console and set the rule for Real Time database like this.

![image](https://user-images.githubusercontent.com/54442328/84475330-90b49300-acb6-11ea-8d61-d20c4dd6c871.png)

**Comment by philipvu30** [comment]
Hi @thawhakyi ,

Please open another ticket and verify your information via our [form](http://verify.inspireui.com/).

Thank you.

**Comment by iamar1k92** [comment]
Realtime and Firestore database has read-write true based rules.
In xcode apple compability is enabled, even on the bundle id the apple login is enabled in apple developer. GoogleServicesInfo.plist is also updated, apple logged in enabled in firebase.



Still cannot login with apple.

**Comment by thawhakyi** [comment]
I can't access verify.inspireui.com.
@philipvu30 

![Annotation 2020-06-12 154844](https://user-images.githubusercontent.com/47990222/84487172-64a00e80-acc4-11ea-811b-477acae632ba.png)

**Comment by philipvu30** [comment]
@thawhakyi 

Use this [link](https://docs.google.com/forms/d/e/1FAIpQLScWOjkAJIWJ9rgb9alopfOqLsJrGKbFM5cCYj1JBfLKLsBblQ/viewform)

**Comment by kalakgl** [comment]
I changed the Real Time database rules, but still have the issue.

**Comment by thawhakyi** [comment]
hi @philipvu30 

I already opened ticket. Pls review it.
can u send config for mylisting theme. thank you.

**Comment by kalakgl** [comment]
@philipvu30 any news about my strange apple login problem?

**Comment by philipvu30** [comment]
Hi @kalakgl ,

I'm sorry that's all I can think of right now. You could check if you have assigned the team for your project yet? Or you can re-download the GoogleService-Info.plist and see if it works.

![image](https://user-images.githubusercontent.com/54442328/84612543-23d60e80-aeeb-11ea-9211-486220cad7a8.png)

**Comment by philipvu30** [comment]
Hi,

I just changed this line and it successfully login. 

Navigate to this file lib/services/woo_commerce.dart and replace SMS -> FB and make sure the endpoint is exactly the same as the image.
![image](https://user-images.githubusercontent.com/54442328/84614873-9c3fce00-aef1-11ea-8ef9-65009cea0169.png)

**Comment by kalakgl** [comment]
@philipvu30 Sorry, but I still have the same error, on a real device and on the simulator.

**Comment by kalakgl** [comment]
@philipvu30 After the "clean" so I could make you an .zip file, then I can't run android simulator.
Now I get these errors:

Compiler message:
../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+2/lib/src/dialogs.dart:272:17: Error: No named parameter with the name 'onRatingChanged'.
                onRatingChanged: (rating) {
                ^^^^^^^^^^^^^^^
../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/smooth_star_rating-1.1.1/lib/smooth_star_rating.dart:24:3: Context: Found this candidate, but the arguments don't match.
  SmoothStarRating({
  ^^^^^^^^^^^^^^^^
lib/widgets/home/banner/banner_map.dart:44:23: Error: A value of type 'PermissionStatus' can't be assigned to a variable of type 'bool'.
 - 'PermissionStatus' is from 'package:location/location.dart' ('../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/location-2.5.4/lib/location.dart').
        _permission = await _locationService.requestPermission();
                      ^

Compiler message:
../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+2/lib/src/dialogs.dart:272:17: Error: No named parameter with the name 'onRatingChanged'.
                onRatingChanged: (rating) {
                ^^^^^^^^^^^^^^^
../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/smooth_star_rating-1.1.1/lib/smooth_star_rating.dart:24:3: Context: Found this candidate, but the arguments don't match.
  SmoothStarRating({
  ^^^^^^^^^^^^^^^^
lib/widgets/home/banner/banner_map.dart:44:23: Error: A value of type 'PermissionStatus' can't be assigned to a variable of type 'bool'.
 - 'PermissionStatus' is from 'package:location/location.dart' ('../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/location-2.5.4/lib/location.dart').
        _permission = await _locationService.requestPermission();
                      ^
Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
build failed.

FAILURE: Build failed with an exception.

* Where:
Script '/Users/ulrik/Development/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 882

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command '/Users/ulrik/Development/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 14s
Exception: Gradle task assembleDebug failed with exit code 1

**Comment by kalakgl** [comment]
Updatd rate_my_app to 0.6.0+3 in pubspec.yaml fixed my android problem.
Now I only have the Apple Login problem.

**Comment by kalakgl** [comment]
Android build, login errors (iOS FB and Google Login are OK):

Google Login:
<img width="471" alt="Skærmbillede 2020-06-15 kl  13 06 55" src="https://user-images.githubusercontent.com/12363496/84674195-6b6b9300-af09-11ea-81b3-81bab47c1ca9.png">

Facebook Login:
<img width="304" alt="Skærmbillede 2020-06-15 kl  13 07 28" src="https://user-images.githubusercontent.com/12363496/84674227-745c6480-af09-11ea-8aff-f317329987e7.png">

**Comment by kalakgl** [comment]
Android FB login fixed.

**Comment by philipvu30** [comment]
Hi,

I tested Apple Login on your source you sent and it is fine on my side. Please have a check again if you  
modify your files correctly with my previous [solution](https://github.com/inspireui/support/issues/3474#issuecomment-643880541)

Because of sensitive information, I cut the email & password input part.
Here is the proof: [3474.mp4.zip](https://github.com/inspireui/support/files/4785321/3474.mp4.zip)

For making the Google Login work for any Android device and any account, you have to add the release key on the Play Store to your FireBase project and replace the file google-services.json in the android folder.

**Comment by kalakgl** [comment]
Yes I can see, but somewhere your sertup are diffenret. Maybe Flutter ?
I have not changed anything beside your solution, but still the Apple login are not working.

**Comment by philipvu30** [comment]
Hi,

Please replace the lib I sent you and see if it works.

> lib/widgets/home/banner/banner_map.dart:44:23: Error: A value of type 'PermissionStatus' can't be assigned to a variable of type 'bool'.
> 
> 'PermissionStatus' is from 'package:location/location.dart' ('../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/location-2.5.4/lib/location.dart').
> _permission = await _locationService.requestPermission();
> ^
> Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
> build failed.

For this issue, please head to the file pubspec.lock, find location and change it from 2.5.4 -> 2.3.5. Then press pub get in pubspec.yaml file again. 

Thank you

**Comment by kalakgl** [comment]
@philipvu30 Thanks, but I got this to work. And the only trouble I have is the iOS Apple Login. Rest is OK incl, Android build.
For Appple Login, I thikn my flutter files are old or new? Because you got this to Work in your computer with my files. But I have no clue how to fix this.

**Comment by philipvu30** [comment]
Hi

Run flutter doctor -v and give me the log.

> And the only trouble I have is the iOS Apple Login

I meant you should try using Apple Login with the files I just gave you. Because I was able to login with those files.

**Comment by kalakgl** [comment]
LOG:
flutter doctor -v 
[✓] Flutter (Channel stable, v1.17.1, on Mac OS X 10.15.5 19F101, locale da-GL)
    • Flutter version 1.17.1 at /Users/ulrik/Development/flutter
    • Framework revision f7a6a7906b (5 weeks ago), 2020-05-12 18:39:00 -0700
    • Engine revision 6bc433c6b6
    • Dart version 2.8.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/ulrik/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.2

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[!] Connected device
    ! No devices available

! Doctor found issues in 1 category.

**Comment by kalakgl** [comment]
@philipvu30 OK I will try with the .zip file you sent and get back to you.
Thanks.

**Comment by kalakgl** [comment]
@philipvu30 With your lib.zip I get this errorlist:

Launching lib/main.dart on iPhone 11 in debug mode...
Running pod install...                                          





Compiler message:
lib/widgets/home/banner/banner_map.dart:44:23: Error: A value of type 'PermissionStatus' can't be assigned to a variable of type 'bool'.
 - 'PermissionStatus' is from 'package:location/location.dart' ('../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/location-2.5.4/lib/location.dart').
        _permission = await _locationService.requestPermission();
                      ^
Running pod install...                                             53,5s
Running Xcode build...
Xcode build done.                                           89,8s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    In file included from /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.m:1:
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    + (instancetype)sharedInstance;
       ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: note: insert '_Nullable' if the pointer may be null
    + (instancetype)sharedInstance;
       ^
                    _Nullable 
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:4:4: note: insert '_Nonnull' if the pointer should never be null
    + (instancetype)sharedInstance;
       ^
                    _Nonnull 
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: warning: block pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: note: insert '_Nullable' if the block pointer may be null
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
                                     _Nullable
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.2.0/ios/Classes/UniLinksPlugin.h:7:33: note: insert '_Nonnull' if the block pointer should never be null
          restorationHandler:(void (^)(NSArray *_Nullable))restorationHandler;
                                    ^
                                     _Nonnull
    2 warnings generated.
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:30: warning: 'requestReview' is only available on iOS 10.3 or newer [-Wunguarded-availability]
        [SKStoreReviewController requestReview];
                                 ^~~~~~~~~~~~~~
    In module 'StoreKit' imported from /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/StoreKit.framework/Headers/SKStoreReviewController.h:22:1: note: 'requestReview' has been marked as being introduced in iOS 10.3 here, but the deployment target is iOS 8.0.0
    + (void)requestReview API_AVAILABLE(ios(10.3), macos(10.14)) API_UNAVAILABLE(watchos) __TVOS_PROHIBITED;
    ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:30: note: enclose 'requestReview' in an @available check to silence this warning
        [SKStoreReviewController requestReview];
                                 ^~~~~~~~~~~~~~
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:6: warning: 'SKStoreReviewController' is only available on iOS 10.3 or newer [-Wunguarded-availability]
        [SKStoreReviewController requestReview];
         ^~~~~~~~~~~~~~~~~~~~~~~
    In module 'StoreKit' imported from /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:2:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/StoreKit.framework/Headers/SKStoreReviewController.h:14:110: note: 'SKStoreReviewController' has been marked as being introduced in iOS 10.3 here, but the deployment target is iOS 8.0.0
    SK_EXTERN_CLASS API_AVAILABLE(ios(10.3), macos(10.14)) API_UNAVAILABLE(watchos) __TVOS_PROHIBITED @interface SKStoreReviewController : NSObject
                                                                                                                 ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/rate_my_app-0.6.0+3/ios/Classes/RateMyAppPlugin.m:15:6: note: enclose 'SKStoreReviewController' in an @available check to silence this warning
        [SKStoreReviewController requestReview];
         ^~~~~~~~~~~~~~~~~~~~~~~
    2 warnings generated.
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:196:43: warning: 'FIRMessagingRemoteMessage' is deprecated: FCM direct channel is deprecated, please use APNs for downstream message handling. [-Wdeprecated-declarations]
    - (void)applicationReceivedRemoteMessage:(FIRMessagingRemoteMessage *)remoteMessage {
                                              ^
    In module 'FirebaseMessaging' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:219:1: note: 'FIRMessagingRemoteMessage' has been explicitly marked deprecated here
    __deprecated_msg(
    ^
    In module 'UIKit' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/sys/cdefs.h:191:48: note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:308:24: warning: 'FIRMessagingRemoteMessage' is deprecated: FCM direct channel is deprecated, please use APNs for downstream message handling. [-Wdeprecated-declarations]
        didReceiveMessage:(FIRMessagingRemoteMessage *)remoteMessage {
                           ^
    In module 'FirebaseMessaging' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:219:1: note: 'FIRMessagingRemoteMessage' has been explicitly marked deprecated here
    __deprecated_msg(
    ^
    In module 'UIKit' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/sys/cdefs.h:191:48: note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:142:30: warning: 'shouldEstablishDirectChannel' is deprecated: FCM direct channel is deprecated, please use APNs channel for downstream message delivery. [-Wdeprecated-declarations]
        [FIRMessaging messaging].shouldEstablishDirectChannel = true;
                                 ^
    In module 'FirebaseMessaging' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:283:56: note: 'shouldEstablishDirectChannel' has been explicitly marked deprecated here
    @property(nonatomic) BOOL shouldEstablishDirectChannel DEPRECATED_MSG_ATTRIBUTE(
                                                           ^
    In module 'UIKit' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:182:64: note: expanded from macro 'DEPRECATED_MSG_ATTRIBUTE'
                #define DEPRECATED_MSG_ATTRIBUTE(s) __attribute__((deprecated(s)))
                                                                   ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:197:52: warning: 'appData' is deprecated [-Wdeprecated-declarations]
      [self didReceiveRemoteNotification:remoteMessage.appData];
                                                       ^
    In module 'FirebaseMessaging' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:226:62: note: 'appData' has been explicitly marked deprecated here
    @property(nonatomic, readonly, strong) NSDictionary *appData DEPRECATED_ATTRIBUTE;
                                                                 ^
    In module 'UIKit' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:180:60: note: expanded from macro 'DEPRECATED_ATTRIBUTE'
            #define DEPRECATED_ATTRIBUTE        __attribute__((deprecated))
                                                               ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:307:1: warning: implementing deprecated method [-Wdeprecated-implementations]
    - (void)messaging:(FIRMessaging *)messaging
    ^
    In module 'FirebaseMessaging' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:253:1: note: method 'messaging:didReceiveMessage:' declared here
    - (void)messaging:(FIRMessaging *)messaging
    ^
    /Users/ulrik/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.13/ios/Classes/FLTFirebaseMessagingPlugin.m:309:63: warning: 'appData' is deprecated [-Wdeprecated-declarations]
      [_channel invokeMethod:@"onMessage" arguments:remoteMessage.appData];
                                                                  ^
    In module 'FirebaseMessaging' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Headers/Public/Firebase/Firebase.h:72:
    /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/FirebaseMessaging/FirebaseMessaging/Sources/Public/FIRMessaging.h:226:62: note: 'appData' has been explicitly marked deprecated here
    @property(nonatomic, readonly, strong) NSDictionary *appData DEPRECATED_ATTRIBUTE;
                                                                 ^
    In module 'UIKit' imported from /Users/ulrik/Data/apps/SisimiutCity_Maj/ios/Pods/Target Support Files/firebase_messaging/firebase_messaging-prefix.pch:2:
    In module 'Foundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:8:
    In module 'CoreFoundation' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
    In module 'Darwin' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:16:
    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.5.sdk/usr/include/AvailabilityMacros.h:180:60: note: expanded from macro 'DEPRECATED_ATTRIBUTE'
            #define DEPRECATED_ATTRIBUTE        __attribute__((deprecated))
                                                               ^
    6 warnings generated.

    Compiler message:
    lib/widgets/home/banner/banner_map.dart:44:23: Error: A value of type 'PermissionStatus' can't be assigned to a variable of type 'bool'.
     - 'PermissionStatus' is from 'package:location/location.dart' ('../../../Development/flutter/.pub-cache/hosted/pub.dartlang.org/location-2.5.4/lib/location.dart').
            _permission = await _locationService.requestPermission();
                          ^
    Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
    build failed.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')
    warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')

Could not build the application for the simulator.
Error launching application on iPhone 11.

**Comment by philipvu30** [comment]
Hi,

I've already sent you my Skype ID. Please check and I'll assist you via TeamViewer

**Comment by philipvu30** [comment]
Closed as solved.
